### PR TITLE
Playlist management

### PR DESCRIPTION
**Pull request by zoef-yt**
[PREVIEW LINK](https://deploy-preview-8--game-plex-zoef.netlify.app/)
[CHECK ME ONLY](https://github.com/zoef-yt/game-plex-react/pull/8/commits/4bbf75e76f6734f4eb9bdff633f057230f93fc25)
In this PR, I added 
- Playlist management and UI for playlist page.

### CODE REVIEW

File: src/Context/HistoryContext/HistoryContext.jsx [hunk 1]
```diff
@@ -4,27 +4,62 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
+	useEffect(() => {
+		if (historyResponse != null && historyResponse.history) {
+			useHistoryList(historyResponse.history);
+		} else {
+			getHistory();
+		}
+	}, [historyResponse]);
+
+	useEffect(() => {
+		console.log(historyError?.response);
```
Comment (iRohitGaur) [inline src/Context/HistoryContext/HistoryContext.jsx]: suggestion: remove `console.log`

**Review by iRohitGaur — APPROVED** [review]
Good work!

UI looks nice. Good to merge 👍🏻

**Comment by santhoshpemmaka** [comment]
No suggestions from my side. Great Work 💯 .

UI looks great. Good to merge 👍 .

**Review by santhoshpemmaka — APPROVED** [review]

File: src/Context/PlaylistContext/PlaylistContext.jsx [hunk 2]
```diff
@@ -4,39 +4,99 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
-	]);
-	return <PlaylistContext.Provider value={{ playlist }}>{children}</PlaylistContext.Provider>;
+	const [playlists, setPlaylist] = useState({ playlists: [] });
+	const [currentVideo, setCurrentVideo] = useState(null);
+	const { response: playlistResponse, operation: playlistServerCall, error: playlistError } = useAxios();
+
+	const token = localStorage.getItem('token');
+
+	useEffect(() => {
+		if (playlistResponse != null && playlistResponse.playlists) {
```
Comment (muskanbagrecha) [inline src/Context/PlaylistContext/PlaylistContext.jsx]: Can use optional chaining

File: src/Context/PlaylistContext/PlaylistContext.jsx [hunk 3]
```diff
@@ -4,39 +4,99 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
+			setPlaylist(playlistResponse);
+		} else {
+			getPlaylists();
+		}
+
+		if (playlistResponse != null && playlistResponse.playlist) {
```
Comment (muskanbagrecha) [inline src/Context/PlaylistContext/PlaylistContext.jsx]: Can use OC.

File: src/Context/PlaylistContext/PlaylistContext.jsx [hunk 4]
```diff
@@ -4,39 +4,99 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
+			const updatedPlaylists = playlists.playlists.map((playlist) => {
```
Comment (muskanbagrecha) [inline src/Context/PlaylistContext/PlaylistContext.jsx:21]: This line of code is difficult to read.

File: src/Pages/Components/Modal/ModalContent/PlaylistModalComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from 'react';
+import { useModal, usePlaylist } from '../../../../Context';
+import './PlaylistModalComponent.css';
+const PlaylistModalComponent = () => {
+	const { closeModal } = useModal();
+	const { playlists, createPlaylist, currentVideo } = usePlaylist();
+	const [playlistTitle, setPlaylistTitle] = useState('');
+
+	const creatNewPlaylistHandler = () => {
+		playlistTitle.length > 1
```
Comment (muskanbagrecha) [inline src/Pages/Components/Modal/ModalContent/PlaylistModalComponent.jsx:10]: Can avoid using ternary when there is nesting in future projects. It makes it easier to read for reviewers.

File: src/Pages/Components/Modal/ModalContent/PlaylistModalComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+						title: playlistTitle,
+					},
+			  }),
+			  currentVideo === null && closeModal(),
+			  setPlaylistTitle(''))
+			: null;
+	};
+	return (
+		<div className='modal-card flex-column align-items-center space-evenly' onClick={(e) => e.stopPropagation()}>
+			{currentVideo != null && (
```
Comment (muskanbagrecha) [inline src/Pages/Components/Modal/ModalContent/PlaylistModalComponent.jsx]: Strict checking

File: src/Pages/Components/Modal/ModalContent/PlaylistModalComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+						})}
+				</div>
+			)}
+
+			<div className='create-playlist-form'>
+				<input className='text-field' type='text' value={playlistTitle} onChange={(e) => setPlaylistTitle(e.target.value)} />
+				<button onClick={creatNewPlaylistHandler} className='btn btn-primary'>
+					Create New Playlist
+				</button>
+				{playlistTitle.length < 1 && <p>*TextField cannot be empty</p>}
```
Comment (muskanbagrecha) [inline src/Pages/Components/Modal/ModalContent/PlaylistModalComponent.jsx:43]: The problem with this is even when the user has just begun typing the error shows up.

File: src/Pages/PlayListPage/PlaylistPage.jsx [hunk 8]
```diff
@@ -1,48 +1,85 @@
...
-			{playlist && playlist.length < 1 ? (
+			{playlists.playlists === null || playlists.playlists.length < 1 ? (
 				<h3 className='text-align-center'>No playlist to show</h3>
 			) : (
 				<div className='video-listing-section'>
-					{playlist.map((singlePlaylist) => {
-						const { _id, videos, name } = singlePlaylist;
-						return <SinglePlaylist key={_id} videos={videos} name={name} />;
-					})}
+					{playlists?.playlists != null &&
```
Comment (muskanbagrecha) [inline src/Pages/PlayListPage/PlaylistPage.jsx]: You can use OC.

File: src/Pages/PlayListPage/PlaylistPage.jsx [hunk 9]
```diff
@@ -1,48 +1,85 @@
...
+	const createNewPlaylistHandler = () => {
+		removeCurrentVideo();
+		user ? openModal('PlaylistModal') : openModal('AuthModal');
+	};
+
 	return (
 		<div className='app-main-content'>
 			<h1 className='text-align-center'>Playlists</h1>
-			{playlist && playlist.length < 1 ? (
+			{playlists.playlists === null || playlists.playlists.length < 1 ? (
```
Comment (muskanbagrecha) [inline src/Pages/PlayListPage/PlaylistPage.jsx:21]: You can make this line shorter.

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Code looks good. Please look into the suggested changes.

File: src/Context/PlaylistContext/PlaylistContext.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Context/PlaylistContext/PlaylistContext.jsx]: done

**Review by muskanbagrecha — APPROVED** [review]
Great work on VL. Animations look superb. Good to merge.

**Pull request merged**